unity-renderer/Assets/Desktop/Scenes/LoginSceneController.cs

```diff
@@ -1,16 +1,37 @@
-using System.Collections;
-using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using DCL.Providers;
 using Login;
+using System;
+using System.Collections;
+using System.Threading;
 using UnityEngine;
+using Environment = DCL.Environment;
 
 public class LoginSceneController : MonoBehaviour
 {
     private LoginHUDController _loginHUDController;
 
+    private CancellationTokenSource cancellationTokenSource;
+
     // Start is called before the first frame update
-    void Start()
+    private IEnumerator Start()
+    {
+        var assetProvider = Environment.i.serviceLocator.Get<IAddressableResourceProvider>();
```

Comment:
You can define
`private Service<IAddressableResourceProvider> addressableService;`
and use addressableService.Ref on this line instead